### PR TITLE
Issue #92 - Refactor Flow and Job to StateMachine and Execution (#128)

### LINKED ISSUE #92: Rename core components
The State Conductor's core components' names overlap with the names of DHF components - leading to confusion as these tool are intended to be used together.

To alleviate these concerns, these components should be renamed.

**"Flows" become "State Machines"** 
- rename state-conductor-flows directory 
- rename affected functions
- rename tracking properties
- rename collections and directories
- rename Rest Service APIs
- Documentation Update

**"Jobs" become "Executions"**
- rename state-conductor-jobs database
- rename affected functions
- rename tracking properties
- rename collections and directories
- rename Rest Service APIs
- rename Data Service APIs
- Documentation Update

This will also better align state conductor terminology with Amazon States Language terms.

### PR DESCRIPTION
**Pull request by aclavio**
Fixes #92

### CODE REVIEW

File: state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/custom-steps-state-machine.asl.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
       "Comment": "runs the dhf 5 flow step",
       "Resource": "/state-conductor/actions/common/dhf/dhf5RunFlowStepAction.sjs",
       "Parameters": {
-        "flowName": "CustomFlow",
+        "name": "CustomFlow",
```
Comment (aclavio) [inline state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/custom-steps-state-machine.asl.json:13]: rollback to keep DHF naming convention of "flowName"

File: state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/missing-dhf-state-machine.asl.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
       "Comment": "runs the dhf 5 flow step",
       "Resource": "/state-conductor/actions/common/dhf/dhf5RunFlowStepAction.sjs",
       "Parameters": {
-        "flowName": "MissingFlow",
+        "name": "MissingFlow",
```
Comment (aclavio) [inline state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/missing-dhf-state-machine.asl.json:13]: rollback to keep DHF naming convention of "flowName"

File: state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/person-state-machine.asl.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
       "Comment": "runs the dhf 5 flow",
       "Resource": "/state-conductor/actions/common/dhf/dhf5RunFlowAction.sjs",
       "Parameters": {
-        "flowName": "PersonFlow",
+        "name": "PersonFlow",
```
Comment (aclavio) [inline state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/person-state-machine.asl.json:13]: rollback to keep DHF naming convention of "flowName"

File: state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/person-steps-state-machine.asl.json [hunk 4]
```diff
@@ -10,7 +10,7 @@
       "Comment": "runs the dhf 5 flow step",
       "Resource": "/state-conductor/actions/common/dhf/dhf5RunFlowStepAction.sjs",
       "Parameters": {
-        "flowName": "PersonFlow",
+        "name": "PersonFlow",
```
Comment (aclavio) [inline state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/person-steps-state-machine.asl.json:13]: rollback to keep DHF naming convention of "flowName"

File: state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/person-steps-state-machine.asl.json [hunk 5]
```diff
@@ -27,7 +27,7 @@
...
```
Comment (aclavio) [inline state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/person-steps-state-machine.asl.json:30]: rollback to keep DHF naming convention of "flowName"

File: state-conductor-modules/src/main/ml-config/databases/state-conductor-jobs-database.json [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {
-  "database-name": "state-conductor-jobs",
+  "database-name": "state-conductor-executions",
```
Comment (freshie) [inline state-conductor-modules/src/main/ml-config/databases/state-conductor-jobs-database.json:2]: the Name of this file should be changed as well

File: state-conductor-modules/src/main/ml-modules/root/state-conductor/state-conductor.sjs [hunk 7]
```diff
@@ -82,40 +82,40 @@ function invokeOrApplyFunction(functionIn, optionsIn) {
...
-function getFlowDocumentFromDatabase(name, databaseId) {
+function getStateMachineFromDatabase(name, databaseId) {
   let resp = invokeOrApplyFunction(
     () => {
-      let flow = getFlowDocument(name);
-      if (!flow) {
-        fn.error(null, 'MISSING-FLOW-FILE', `Cannot find a a flow file with the name: ${name}`);
+      let stateMachine = getStateMachine(name);
+      if (!stateMachine) {
+        fn.error(null, 'MISSING-STATE_MACHINE-FILE', `Cannot find a a stateMachine file with the name: ${name}`);
```
Comment (freshie) [inline state-conductor-modules/src/main/ml-modules/root/state-conductor/state-conductor.sjs:116]: I think there is a type O from before with two "a" next to each other

**Review by freshie — CHANGES_REQUESTED** [review]
added .DS_Store to git ignore

File: state-conductor-cpf/src/main/ml-modules/root/state-conductor/cpf/state-conductor-work-action.sjs [hunk 8]
```diff
@@ -11,16 +11,16 @@ var transition;
 if (cpf.checkTransition(uri, transition)) {
   try {
     xdmp.trace(sc.TRACE_EVENT, `state-conductor-work-action for "${uri}"`);
-    const continueJob = sc.processJob(uri);
-    if (continueJob) {
-      // continue cpf processing - continuing the current flow or any others that apply
+    const continueExecution = sc.processExecution(uri);
```
[the remark this answers is not in the document]
Comment (freshie) [reply state-conductor-cpf/src/main/ml-modules/root/state-conductor/cpf/state-conductor-work-action.sjs:14]: Will have another commit to remove cpf stuff.

**Review by freshie — APPROVED** [review]

**Comment by freshie** [comment]
i'm going to push changes from my own branch as this one isn't updating.

**Pull request closed without merging**